assets/javascripts/discourse/services/d-templates.js

```diff
@@ -0,0 +1,102 @@
+import Service, { inject as service } from "@ember/service";
+import { getOwner } from "discourse-common/lib/get-owner";
+import DTemplatesModalForm from "../components/d-templates/modal/form";
+import { replaceVariables } from "../../lib/replace-variables";
+import TextareaManipulator from "../../lib/textarea-manipulator";
+import extractVariablesFromComposerModel from "../../lib/variables-composer";
+
+export default class DTemplatesService extends Service {
+  @service appEvents;
+  @service modal;
+  @service site;
+
+  showComposerUI() {
+    const onInsertTemplate = this.#insertTemplateIntoComposer;
```

Comment:
That method will need to be bound, either with `@bind` (but then it can't be `#` private, because the decorator isn't compatible with that; you'd have to just prefix the name with `_`) or here:

```suggestion
    const onInsertTemplate = this.#insertTemplateIntoComposer.bind(this);
```